### PR TITLE
SubmitMillauToRialtoMessage subcommand in substrate-relay

### PR DESCRIPTION
**Pull request by svyatonik**
related #219

### CODE REVIEW

File: relays/millau-client/src/lib.rs [hunk 1]
```diff
@@ -42,6 +46,68 @@ impl Chain for Millau {
...
+				frame_system::CheckTxVersion::<millau_runtime::Runtime>::new(),
+				frame_system::CheckGenesis::<millau_runtime::Runtime>::new(),
+				frame_system::CheckEra::<millau_runtime::Runtime>::from(sp_runtime::generic::Era::Immortal),
+				frame_system::CheckNonce::<millau_runtime::Runtime>::from(signer_nonce),
+				frame_system::CheckWeight::<millau_runtime::Runtime>::new(),
+				pallet_transaction_payment::ChargeTransactionPayment::<millau_runtime::Runtime>::from(0),
+			),
+			(
+				millau_runtime::VERSION.spec_version,
+				millau_runtime::VERSION.transaction_version,
```
Comment (HCastano) [inline relays/millau-client/src/lib.rs:73]: Could these be configurable via the CLI, or is that a bad idea?

File: relays/substrate/src/cli.rs [hunk 2]
```diff
@@ -38,6 +39,52 @@ pub enum Command {
...
+		message: ToRialtoMessage,
+		/// Delivery and dispatch fee.
+		#[structopt(long)]
+		fee: bp_millau::Balance,
+	},
+}
+
+arg_enum! {
+	#[derive(Debug)]
+	/// All possible messages that may be delivered from to Rialto chain.
```
Comment (HCastano) [inline relays/substrate/src/cli.rs:64]:
```suggestion
	/// All possible messages that may be delivered to the Rialto chain.
```

File: relays/substrate/src/cli.rs [hunk 3]
```diff
@@ -38,6 +39,52 @@ pub enum Command {
 		#[structopt(flatten)]
 		prometheus_params: PrometheusParams,
 	},
+	/// Submit message to given Rialto -> Millau lane.
+	SubmitMillauToRialtoMessage {
+		#[structopt(flatten)]
+		millau: MillauConnectionParams,
+		#[structopt(flatten)]
+		millau_sign: MillauSigningParams,
```
Comment (HCastano) [inline relays/substrate/src/cli.rs:47]: Is there any way to give example defaults here?

File: relays/substrate/src/cli.rs [hunk 4]
```diff
@@ -38,6 +39,52 @@ pub enum Command {
...
+		#[structopt(flatten)]
+		rialto_sign: RialtoSigningParams,
```
Comment (HCastano) [inline relays/substrate/src/cli.rs:49]: Same question here

**Review by HCastano — APPROVED** [review]

**Comment by HCastano** [comment]
CI errors don't look like they're related to your PR. I think we need to bump Substrate

File: relays/millau-client/src/lib.rs (hunk 1 above)
Comment (svyatonik) [reply relays/millau-client/src/lib.rs:73]: It has to be solved as a part of #445 . But generally speaking, encoding/tx format/... may change between versions, and you should only create transactions of the runtime you are compiled with. Unless you're sure that nothing has actually changed, which we can't prove => better not to provide spec versions using cli. Right now all relay transactions are created that way.

File: relays/substrate/src/cli.rs (hunk 3 above)
Comment (svyatonik) [reply relays/substrate/src/cli.rs:47]: I remember Tomek has had some problems with default values when he worked on eth2sub/sub2eth deployments, which he has mentioned in #162. So I haven't added any default values (except for prometheus cfg).

Having default values is imo better than not having them. But if it causes problems, then let's add all arguments explicitly.

**Pull request merged**